orbit/radius.py

```diff
@@ -468,17 +469,24 @@ def form_respond():
     if not (student_id := rocket.body_args_query('student_id')):
         rocket.msg('you must provide a student id')
         return form_respond()
-    query = (db.Registration
-             .delete()
-             .where(db.Registration.student_id == student_id)
-             .returning(db.Registration))
-    if not (registration := next(iter(query.execute()), None)):
+
+    pw_entropy = os.urandom(config.num_bytes_entropy_for_pw)
+    password = base64.b64encode(pw_entropy, altchars=b'-_')
+    salt = bcrypt.gensalt()
+    pwdhash = bcrypt.hashpw(password, salt).decode('utf-8')
+
+    query = (db.User
+             .update({db.User.pwdhash: pwdhash})
+             .where((db.User.student_id == student_id) &
```

Comment:
tl;dr: You cannot use `and`, you must use `&`

long version:
To create this nice syntax peewee uses operator overloading (i.e. `db.User.sudent_id == whatever` evaluates to a constraint object between that field and that value because the peewee fields have an `__eq__` function that provides an overload for `==`).

You cannot overload the behavior of the `and` operator in python and instead the best you can do is overload `&` by defining `__and__`.

This is because the `and` keyword has to be a language feature to do short-circuit evaluation. If you overloaded `and` the compiler would need to evaluate both sides of the expression before it could pass their values to your overloaded function instead of only evaluating the first and potentially skipping the second if the first fails.

This is one of the many pitfalls present in C++ where you can overload `operator&&` but you really really really never should because it breaks short circuit: https://en.cppreference.com/w/cpp/language/operators
>  * The overloads of operators && and || lose short-circuit evaluation. 
>  * &&, ||, and , (comma) lose their special sequencing properties when overloaded and behave like regular function calls even when they are used without function-call notation.